pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/elastic/ElasticSearchService.java

```diff
@@ -372,4 +373,57 @@ public static void deleteParticipantDocument(String docId, String index) {
                     .formatted(docId, index), e);
         }
     }
+
+    /**
+     * Get the DSM object for a single participant from ES as a Map
+     * or throw an exception if the participant or DSM object is not found
+     * @param realm the realm
+     * @param participantIndex the participant index
+     * @param shortId the shortId of the participant
+     * @return the DSM participant object
+     */
+    public static Map<String, Object> getDsmForSingleParticipant(String realm, String participantIndex, String shortId) {
+        Map<String, Map<String, Object>> ptpData;
+        try {
+            ptpData = ElasticSearchUtil.getSingleParticipantFromES(realm, participantIndex, shortId);
+        } catch (Exception e) {
+            throw new DsmInternalError("ES threw exception for search of shortId: " + shortId, e);
+        }
+
+        if (ptpData.size() > 1) {
+            String msg = String.format("ES returned %d results for participant shortId %s", ptpData.size(), shortId);
+            throw new DsmInternalError(msg);
+        }
+        if (ptpData.size() == 0) {
+            throw new DSMBadRequestException("Invalid participant ID " + shortId);
+        }
+        Map<String, Object> ptp = ptpData.values().stream().findFirst().orElseThrow();
+        Map<String, Object> dsm = (Map<String, Object>) ptp.get("dsm");
+        if (dsm == null || dsm.isEmpty()) {
+            throw new DsmInternalError("ES returned empty dsm object for shortId " + shortId);
+        }
+        return dsm;
```

Comment:
You are correct, so it seems like this method is more confusing and not a great and useful tool for getting data out of ES, I searched for a method to do this similarly and didn't find anything, so I created a new method to search by short id  in the `ElasticSearch` class and edited this to use that.

Reply:
I think perhaps I was not clear in my message... these methods are fine, they just want to do one more step: return an ES object rather than a Map. (See other methods in this class.)
The ElasticSearch class is not a good place to put new code, in my opinion.